profiles/common/modules/features/nexteuropa_trackedchanges/nexteuropa_trackedchanges.admin.inc

```diff
@@ -0,0 +1,700 @@
+    );
+  }
+  else {
+    $form['table_status'] = array(
+      '#markup' => t('The rebuild of the logs table is under way.'),
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * Form submit callback.
+ *
+ * @see nexteuropa_trackedchanges_rebuild_status()
+ */
+function nnexteuropa_trackedchanges_rebuild_elapse_time_submit($form, &$form_state) {
+  $values = $form_state['values'];
+  $elapse_time_value = (!empty($values['nexteuropa_trackedchanges_rebuild_elapse_time'])) ? $values['nexteuropa_trackedchanges_rebuild_elapse_time'] : 0;
+
+  variable_set('nexteuropa_trackedchanges_rebuild_elapse_time', $elapse_time_value);
+}
+
+/**
+ * Form validate callback.
+ *
+ * @see nexteuropa_trackedchanges_rebuild_status()
+ */
+function nexteuropa_trackedchanges_rebuild_elapse_time_validate($form, &$form_state) {
+  $values = $form_state['values'];
+  $elapse_time_value = $values['nexteuropa_trackedchanges_rebuild_elapse_time'];
+  if (!is_numeric($elapse_time_value) || ($elapse_time_value < 0)) {
+    form_set_error('nexteuropa_trackedchanges_rebuild_elapse_time',
+      t('The "refresh frequency" field must be a positive number.'));
+  }
+}
+
+/**
+ * Submit form call back triggering the netc_info table.
+ *
+ * @see nexteuropa_trackedchanges_rebuild_status()
+ */
+function nexteuropa_trackedchanges_table_rebuilt_submit($form, &$form_state) {
+
+  // Empty the table and "lock" it.
+  variable_set('nexteuropa_trackedchanges_rebuild_lock', 1);
+  // It is a rebuild, then we flush the tables.
+  _nexteuropa_trackedchanges_clean_db();
+
+  $operations = array();
+  // The 2nd argument is useless in this case because the default Batch API
+  // process fulfils function requirements.
+  $operations[] = array('_nexteuropa_trackedchanges_revisions_scanning_process', array(''));
+  $operations[] = array('_nexteuropa_trackedchanges_entity_scanning_process', array(''));
+  $operations[] = array('_nexteuropa_trackedchanges_table_rebuilt_process', array(''));
+
+  $batch = array(
+    'operations' => $operations,
+    'finished' => '_nexteuropa_trackedchanges_table_rebuilt_finished',
+    'title' => t('Processing the rebuild of the tracked change table'),
+    'init_message' => t('Content scanning is starting.'),
+    'progress_message' => t('Processed @current out of @total.'),
+    'error_message' => t('The rebuild has encountered an error.'),
+  );
+  batch_set($batch);
+}
+
+/**
+ * Finished callback of the batch rebuilding the "Info" table.
+ *
+ * The "Info" table is "nexteuropa_trackedchanges_info".
+ *
+ * @see nexteuropa_trackedchanges_table_rebuilt()
+ */
+function _nexteuropa_trackedchanges_table_rebuilt_finished($success, $results, $operations) {
+  // Unlock the table to allow a new rebuild if it failed.
+  variable_set('nexteuropa_trackedchanges_rebuild_lock', 0);
+  variable_set('nexteuropa_trackedchanges_last_rebuild_time', REQUEST_TIME);
+  if ($success) {
```

Comment:
The condition imbrications are correct and reflect the process reality.

Furthermore, there is no default message possible. The 3 possibilities are exclusive and required a special displayed message.